### ISSUE TITLE
Switch from flow to typescript

### ISSUE THREAD

**Issue by machour** [gitpoint/git-point#772]
As we were going through #770 with @housseindjirdeh, we both emitted the desire of ditching flow in favor of typescript.

I personally struggled with flow while trying to enforce types, using VS Code. Validation sometimes takes a few seconds, and the whole thing seems a bit sketchy to me. 

Of course, this is up for debate :)

**Comment by housseindjirdeh** [comment]
I also have a preference for TS for a few reasons (one being a smoother editor experience). Thanks for logging this mate - and happy to hear anybody else's opinions on this as well 💬

**Comment by G2Jose** [comment]
Would love to see this as well! Lmk if we're cool with me creating an initial PR for this.

**Comment by alejandronanez** [comment]
Whew, this is quite an old issue 😅.

My 2 cents about this, what specific problems would typescript

**Comment by chinesedfan** [comment]
Agreed with @alejandronanez. The main difference between them is that flow supports to be used progressively and it is already there. If created a new project, I will choose Typescript. And I don't think we can gain much benefits and have enough time to do that.
